backend/src/handlers/open_content_handler.go

```diff
@@ -109,3 +111,36 @@ func (srv *Server) handleGetUserFavoriteOpenContent(w http.ResponseWriter, r *ht
 	meta := models.NewPaginationInfo(page, perPage, total)
 	return writePaginatedResponse(w, http.StatusOK, favorites, meta)
 }
+
+func (srv *Server) handleToggleFeatureLibrary(w http.ResponseWriter, r *http.Request, log sLog) error {
+	var reqBody models.OpenContentParams
+	if err := json.NewDecoder(r.Body).Decode(&reqBody); err != nil {
+		return newJSONReqBodyServiceError(err)
+	}
+	defer r.Body.Close()
+
+	userID := r.Context().Value(ClaimsKey).(*Claims).UserID
+	facilityID := r.Context().Value(ClaimsKey).(*Claims).FacilityID
+	contentID, err := strconv.Atoi(r.PathValue("id"))
+	if err != nil {
```

Comment:
if we have this check at the very beginning (since the rest of the handler depends on it), it can fail first and we can return error before opening the request body, etc